CONVENTIONS.md

```diff
@@ -0,0 +1,55 @@
+
+## Flow Control
+
+Wherever it makes sense the ternary operator ought to be preferred over an if statement for the sake of brevity, but never at the expense of readability. If the statement contains a long if condition it should not be turned into a ternary operator because the line will be too long and the statement will lose readability. If the statement contains a short if else statement and can be expressed simply as a ternary operator, that ought to be preferred.
+
+## Enums
+
+Enum values that are referenced internally (values referenced within methods associated with their own enum definition) must always use `Enum.VALUE` never simply `VALUE`.
+
+## Mixins
+
+Regular mixin classes must be named the same name as the target class they are being mixed into, with "Mixin" appended to the end of the class name. As an example, a mixin class into the class `MinecraftServer` should be named `MinecraftServerMixin`.
+
+Likewise, accessor mixin classes must be named the same name as the target class they are being mixed into, with "Accessor" appended to the end of the class name. As an example, the accessor class into the class `MinecraftServer` should be named `MinecraftServerAccessor`.
+
+"Duck interfaces" (Interfaces that are implemented onto classes using mixins) should be named the same name as the target class that implements the interface, with a series of adjectives at the front of the class name describing the contents or the implementation use case of the interface, along with "Hooks" at the end of the class name. For example, a duck interface implemented by `PlayerEntity` through mixins in order to add a mana player attribute should be named `ManaPlayerEntityHooks`.
+
+Mixins that shadow final fields must use the `@Final` annotation provided by the mixin library.
+
+Mixin injection and redirect methods must be named based upon the specific function of the injection or redirection, not the method that they are injecting into or redirecting. For example a method inject that registers commands on the dedicated server ought to be named `registerCommands` not the name of the target method, in this case `setupServer`.
+
+If a shadowed method needs to be prefixed because of a conflict it must use the prefix `minecraft$` for methods in minecraft, `[namespace]$` for projects with a different namespace (which should always match the mod id), or if neither of those are available `[package name]$`. Take for example a situation where the shadowed method `fromJson` conflicts with a new `fromJson` method. The shadowed method should be renamed to `minecraft$fromJson` and annotated as follows `@Shadow(prefix = "minecraft$")`.
+
+All mixin classes soft implementing interfaces must use the prefix `fabric$` and the `@Implements` annotation from the mixin library. For example, take a class that soft implements the interface `Identifiable`, it must be annotated as follows `@Implements(@Interface(iface = Identifiable.class, prefix = "fabric$"))`.
```

Comment:
`fabric$` specifier is probably not right for every context. This specific case probably belongs in a future pull request.